lambda/translmod.ml

```diff
@@ -919,40 +919,134 @@ let required_globals ~flambda body =
   Translprim.clear_used_primitives ();
   required
 
+let add_arg_block_to_module_block primary_block_lam size restr =
+  let primary_block_id = Ident.create_local "*primary-block*" in
+  let arg_block_id = Ident.create_local "*arg-block*" in
+  let arg_block_lam =
+    apply_coercion Loc_unknown Strict restr (Lvar primary_block_id)
+  in
+  let get_field i = Lprim (mod_field i, [Lvar primary_block_id], Loc_unknown) in
+  let all_fields = List.init size get_field @ [Lvar arg_block_id] in
+  let arg_block_field = size in
+  let new_size = size + 1 in
+  Llet(Strict, layout_module, primary_block_id, primary_block_lam,
+       Llet(Strict, layout_module, arg_block_id, arg_block_lam,
+            Lprim(Pmakeblock(0, Immutable, None, alloc_heap),
+                  all_fields,
+                  Loc_unknown))),
+  new_size,
+  Some arg_block_field
+
+let add_parameters lam params =
```

Comment:
this special-purpose function is unsuitable as a top-level function, especially there is another function below called "has_parameters". At least call it `add_runtime_parameters`

Reply:
Yeah, that's a better name for sure, changing it